float64 distance
float64 theta
float64 phi
float64 dist_center_plane